src/lib/registration/addon.rb

```diff
@@ -80,6 +80,27 @@ def registered_not_installed
         end
       end
 
+      # Returns passed addons sorted with all dependencies ordered that it can be
+      # registered from first to last ( so no dependencies for first ).
+      # @param list [Array<Addon>] of addons
+      def registration_order(list)
+        to_process = list.dup
+        result = []
+
+        loop do
+          break if to_process.empty?
+          next_addon = to_process.find do |addon|
+            addon.depends_on.nil? || !to_process.include?(addon.depends_on)
+          end
+          raise "circular dependencies found in addons #{to_process.inspect}" unless next_addon
```

Comment:
`Report.Error` would be probably more user friendly, but hopefully this never happens so let's make it simple...

Reply:
Well, I hope we catch it quickly in openQA when this happen, so I hope not needed. Also Report is useless as result is that it won't work.

Reply:
The problem is that it might happen *after* the release, few months later... But anyway as this is really a corner case let's use this simple solution...